### PR TITLE
performance optimisation: avoid map regrow calls

### PR DESCRIPTION
**Pull request by chirayu**

### CODE REVIEW

File: runtime/context.go [hunk 1]
```diff
@@ -205,9 +206,10 @@ func WithScopeTags(ctx context.Context, newFields map[string]string) context.Con
 
 // GetScopeTagsFromCtx returns the tag info extracted from context.
 func GetScopeTagsFromCtx(ctx context.Context) map[string]string {
-	tags := make(map[string]string)
+	var tags map[string]string
```
Comment (hbhoyar-uber) [inline runtime/context.go:209]: Same comment as above

File: runtime/context.go [hunk 2]
```diff
@@ -119,9 +119,10 @@ func WithEndpointRequestHeadersField(ctx context.Context, requestHeaders map[str
 
 // GetEndpointRequestHeadersFromCtx returns the endpoint request headers, if it exists on context
 func GetEndpointRequestHeadersFromCtx(ctx context.Context) map[string]string {
-	requestHeaders := make(map[string]string)
+	var requestHeaders map[string]string
```
[the remark this answers is not in the document]
Comment (chirayu) [reply runtime/context.go:122]: good catch..

File: runtime/context.go [hunk 3]
```diff
@@ -119,9 +119,10 @@ func WithEndpointRequestHeadersField(ctx context.Context, requestHeaders map[str
 
 // GetEndpointRequestHeadersFromCtx returns the endpoint request headers, if it exists on context
 func GetEndpointRequestHeadersFromCtx(ctx context.Context) map[string]string {
-	requestHeaders := make(map[string]string)
+	requestHeaders := make(map[string]string, 0)
```
Comment (hbhoyar-uber) [inline runtime/context.go:122]: nit: 0 is default length hence not required to specify

File: runtime/context.go [hunk 4]
```diff
@@ -205,9 +206,10 @@ func WithScopeTags(ctx context.Context, newFields map[string]string) context.Con
 
 // GetScopeTagsFromCtx returns the tag info extracted from context.
 func GetScopeTagsFromCtx(ctx context.Context) map[string]string {
-	tags := make(map[string]string)
+	tags := make(map[string]string, 0)
```
Comment (hbhoyar-uber) [inline runtime/context.go:209]: nit: same as above

**Review by hbhoyar-uber — APPROVED** [review]

File: runtime/context.go (hunk 3 above)
Comment (chirayu) [reply runtime/context.go:122]: See https://pkg.go.dev/builtin#make . Go allocates a small size, if it is not specified.

**Pull request merged**